docs/code/MlNetHighLevelConcepts.md

```diff
@@ -0,0 +1,151 @@
+# ML.NET high-level concepts
+
+In this document, we give a brief overview of the ML.NET high-level concepts. This document is mainly intended to describe the *model training* scenarios in ML.NET, since not all these concepts are relevant for the more simple scenario of *prediction with existing model*.
+
+## List of high-level concepts
+
+This document is going to cover the following ML.NET concepts:
+
+- *Data*, represented as an `IDataView` interface.
+  - In ML.NET, data is very similar to a SQL view: it's a lazily-evaluated, immutable, cursorable, heterogenous, schematized dataset. 
+  - An excellent document about the data interface is [IDataView Design Principles](IDataViewDesignPrinciples.md).
+- *Transformer*, represented as `ITransformer` interface.
+  - In one sentence, a transformer is a component that takes data, does some work on it, and return new 'transformed' data.
+  - For example, you can think of a machine learning model as a transformer that takes features and returns predictions.
+- *Data reader*, represented as an `IDataReader<T>` interface.
```

Comment:
Maybe put this right after "Data"?

Reply:
My narrative will get broken this way. I'll keep the order

---
In reply to: [219392460](https://github.com/dotnet/machinelearning/pull/960#discussion_r219392460) [](ancestors = 219392460)